iceoryx_utils/test/moduletests/test_posix_mutex.cpp

```diff
@@ -49,30 +44,45 @@ class Mutex_test : public Test
         }
     }
 
-    iox::posix::mutex sut{false};
+    iox::posix::mutex sutNonRecursive{false};
+    iox::posix::mutex sutRecursive{true};
+    iox::units::Duration m_watchdogTimeout = 5_s;
+    Watchdog m_deadlockWatchdog{m_watchdogTimeout};
 };
 
-TEST_F(Mutex_test, TryLockWithNoLock)
+TEST_F(Mutex_test, TryLockAndUnlockWithNonRecursiveMutexReturnTrue)
 {
-    EXPECT_THAT(sut.try_lock(), Eq(true));
-    EXPECT_THAT(sut.unlock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.try_lock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.unlock(), Eq(true));
 }
 
-TEST_F(Mutex_test, TryLockWithLock)
+
+#ifndef _WIN32
+TEST_F(Mutex_test, TryLockWithNonRecursiveMutexReturnsFalseAfterLock)
 {
-    EXPECT_THAT(sut.lock(), Eq(true));
-    EXPECT_THAT(sut.try_lock(), Eq(false));
-    EXPECT_THAT(sut.unlock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.lock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.try_lock(), Eq(false));
+    EXPECT_THAT(sutNonRecursive.unlock(), Eq(true));
 }
+#endif
 
-TEST_F(Mutex_test, LockAndUnlock)
+TEST_F(Mutex_test, LockAndUnlockWithNonRecursiveMutexReturnsTrue)
 {
-    EXPECT_THAT(sut.lock(), Eq(true));
-    EXPECT_THAT(sut.unlock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.lock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.unlock(), Eq(true));
 }
 
+TEST_F(Mutex_test, RepeatedLockAndUnlockWithNonRecursiveMutexReturnsTrue)
+{
+    EXPECT_THAT(sutNonRecursive.lock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.unlock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.lock(), Eq(true));
+    EXPECT_THAT(sutNonRecursive.unlock(), Eq(true));
+}
+
+#if !defined(_WIN32) && !defined(QNX) && !defined(QNX__) && !defined(__QNX__) && !defined(__APPLE__)
```

Comment:
I'm getting a bit confused, so, we do **NOT** want to execute this test if the platform is Win, QNX or Apple. Correct?